columnflow/columnar_util.py

```diff
@@ -1201,16 +1201,14 @@ def fill_hist(
             if name not in data:
                 raise ValueError(f"missing data for histogram axis '{name}'")
 
-    # create numpy views for all data arrays
-    data = {name: np.asarray(data[name]) for name in data}
+    # create awkward views for all data arrays
+    data = {name: ak.Array(data[name]) for name in data}
 
     # correct last bin values
     for ax in correct_last_bin_axes:
         right_egde_mask = data[ax.name] == ax.edges[-1]
         if np.any(right_egde_mask):
-            # work on a copy to not change input values in-place
-            data[ax.name] = data[ax.name].copy()
-            data[ax.name][right_egde_mask] -= ax.widths[-1] * 1e-5
+            data[ax.name] = data[ax.name] - ax.widths[-1] * 1e-5
```

Comment:
GitHub does not let me make a suggestion here for some reason, but the loop should be:

```
# correct last bin values
for ax in correct_last_bin_axes:
    right_egde_mask = ak.flatten(data[ax.name], axis=None) == ax.edges[-1]
    if np.any(right_egde_mask):
        data[ax.name] = data[ax.name].copy()
        flat_np_view(data[ax.name])[right_egde_mask] -= ax.widths[-1] * 1e-5
```

Reply:
ok, I see now the problem with what I had. It's been changed.

Reply:
true, sorry! Fixed that now